src/query.c

```diff
@@ -1077,7 +1077,6 @@ typedef IndexIterator **IndexIteratorArray;
  * should be performed. If true, the string remains case-sensitive.
  */
 static void tag_strtolower(char *str, size_t *len, int caseSensitive) {
-  size_t origLen = *len;
```

Comment:
To avoid redundant memory accesses, consider storing the updated length in a local variable and updating `*len` only once at the end.

Reply:
OK. Done.